frontend/src/components/RotateCredentials.vue

```diff
@@ -0,0 +1,429 @@
+<!--
+SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+<template>
+  <action-button-dialog
+    :shoot-item="shootItem"
+    :loading="isLoading"
+    :disabled="isDisabled"
+    @dialog-opened="startDialogOpened"
+    ref="actionDialog"
+    :caption="componentTexts.caption"
+    :tooltip="tooltip"
+    :icon ="icon"
+    width="700"
+    :button-text="componentTexts.buttonText"
+    :confirm-required="true"
+    :confirm-button-text="confirmButtonText">
+    <template v-slot:actionComponent>
+      <v-row >
+        <v-col>
+          <div class="py-4 text-h5 pt-0 pb-3">{{componentTexts.heading}}</div>
+          <v-alert v-if="mode === 'start'" type="info" outlined dense>Note: This rotation operation is split into two steps. This step will <strong>prepare</strong> the rotation.</v-alert>
+          <v-alert v-if="mode === 'complete'" type="info" outlined dense>Note: This rotation operation is split into two steps. This step will <strong>complete</strong> the rotation.</v-alert>
+          <strong>Actions performed in this step</strong>
+          <ul>
+            <li
+              v-for="action in componentTexts.actions"
+              :key="action">
+              {{action}}
+            </li>
+          </ul>
+          <v-checkbox
+            v-model="maintenance"
+            label="Perform this operation in the maintenance time window"
+            :disabled="isMaintenanceDisabled"
+            :hint="maintenanceHint"
+            persistent-hint>
+          </v-checkbox>
+          <div class="mt-3">Type <span class="font-weight-bold">{{shootName}}</span> below to confirm the operation.</div>
+        </v-col>
+      </v-row>
+    </template>
+  </action-button-dialog>
+</template>
+
+<script>
+import ActionButtonDialog from '@/components/dialogs/ActionButtonDialog'
+import { addShootAnnotation } from '@/utils/api'
+import { rotationTypes } from '@/utils/credentialsRotation'
+import { SnotifyPosition } from 'vue-snotify'
+import { shootItem } from '@/mixins/shootItem'
+import { errorDetailsFromError } from '@/utils/error'
+import { mapGetters } from 'vuex'
+import includes from 'lodash/includes'
+import get from 'lodash/get'
+import compact from 'lodash/compact'
+
+export default {
+  name: 'rotate-credentials',
+  components: {
+    ActionButtonDialog
+  },
+  props: {
+    text: {
+      type: Boolean
+    },
+    type: {
+      type: String,
+      required: false
+    }
+  },
+  data () {
+    return {
+      actionTriggered: false,
+      maintenance: false
+    }
+  },
+  mixins: [shootItem],
+  computed: {
+    ...mapGetters([
+      'isAdmin'
+    ]),
+    mode () {
+      if (!this.completionOperation) {
+        return 'rotate'
+      }
+      if (this.operation === this.completionOperation) {
+        return 'complete'
+      }
+      return 'start'
```

Comment:
modes in uppercase? similar like we have it with the constants for the authentication modes for user and application credential

Reply:
good idea